test/fixtures/google/tokenized_card.json

```diff
@@ -0,0 +1,5 @@
+{
+  "signature": "MEUCIQDzgQkY/Lj42FoZC9wN44Uz0r6uDttc68k2c9oZCzNJPwIgSRukEx30uMjqzFswekoarAtO9p1/E221hNmVuHzDW+M\u003d",
+  "protocolVersion": "ECv1",
```

Comment:
This payment token structure looks very different from the [one we have for Android Pay](https://github.com/bdewater/r2d2/blob/cc60cd651c0d13d548f747c4e82b6fee117a3952/test/fixtures/google/card.json). If this is a structure we can rely on, it would be interesting to provide a single entry point into r2d2 that determines which token path to use to decrypt vs. having the user have to peek into the structure to determine which token type it is.

Not sure if this is a real world scenario or not? Definitely would still keep very explicit Google vs. Android token types, just with a thin routing convenience on top if we wanted to provide this.

Reply:
Android Pay was unversioned and [Google's documentation](https://developers.google.com/pay/api/web/resources/payment-data-cryptography) says in this case to assume `ECv0`. A factory method would definitely be nice to abstract this concern. I can add this somewhere this week.